### DISCUSSION TITLE
Give access to sponsors-only repositories for one-time tiers

### DISCUSSION THREAD

**Discussion by alexandreruban** [community/community#14379] [Other Feature Feedback, Questions, & Ideas]
Hello Github team! 👋

Currently, it is only possible to grant a sponsor access to a private repository for *monthly tiers*. It would be nice to be able to do the same thing for *one-time tiers*.

#### Example use case

I wrote an open-source tutorial. Even if it is open-source, it would be nice to make a little money from it. I proposed giving access to the tutorial's source code to the sponsors who send me a one-time payment of 25$. It feels strange to ask for recurring payment for this kind of open-source contribution, so I kept the *one-time tier* and manually give the sponsors access to the repository.

What do you think? Is there any reason to prevent contributors from giving access to private repositories for *one-time tiers*?

**Comment by jlord** [comment]
Hi! This is great feedback. We are planning to do another round of work on sponsors-only-repos after it'd been out in the wild a bit and this use case makes a lot of sense.

**Reply by alexandreruban** [reply]
@jlord any news on this potential feature? 😇

**Comment by Rey-Hax** [comment]
Also, it seems like a tier can grant access to only one repo. I think it would be great if you could have a tier grant access to a list of repos. I'm looking to use GitHub sponsors in conjunction with the [Unity Asset Store](https://assetstore.unity.com/) where sponsors can have access to assets for a smaller fee, but reoccurring as oppose to a one time fee on the asset store. I have a few assets in production and I think it would be cool to have one tier that grants access to all repos of my assets.

**Reply by LeonardoCiaccio** [reply]
I support this post, I have a similar problem, I have several private repositories and I would like to make access only to sponsors. Really strange that you have not designed such access, but I am sure you @github-staff  will provide it soon, you are great.

**Comment by fboeller** [comment]
In support of this idea!

My use case is similar: I have a series of blog post freely available on my blog and offer access to the full code as a running application by sponsoring on GitHub. Most people likely just want to pay once and have the code.

**Comment by reeseschultz** [comment]
I, too, request that monthly and one-time tiers be able to grant read access to lists of sponsors-only repos, not just one repo per tier.

I want to offer a software suite via GitHub Sponsors, but that won't be realistically possible (without hacks) unless this request is fulfilled. I have no doubt it must be soon, because otherwise there's no simple way to compete with other marketplaces such as https://itch.io and the Unity Asset Store. If I try licensing a C# library via monthly tier to Unity users, they'll just laugh in my face, because they've been conditioned to make large one-time payments for perpetual licensing. They'll tell me to use the Asset Store, even though said library *could* be used with something like MonoGame. I personally don't want to distribute via the Asset Store, and I don't want to complicate my EULA by codifying different providers and following the union of all of their terms and conditions.

In other words, I'm really hopeful that GitHub Sponsors could be my one-stop shop, especially if [sponsors-only actions](https://github.com/github-community/community/discussions/12563) could be addressed as well. Distributing code via GitHub is a dream. I can direct, say, Unity users to use their package manager, which is compatible with Git and GitHub by extension, and I can direct MonoGame folks to submodules, which is much more transparent and fluid than NuGet in my humble opinion.

**Comment by wnelson03** [comment]
Yes please, would really appreciate this be added soon :)

**Reply by mazkdevf** [reply]
+ 1 UP!

**Comment by Andersama** [comment]
Seems like this shouldn't be particularly difficult to write in as a feature, is there a holdup for some other reason?

I'd like to add if it was just forgotten, consider a "cumulative" option or "cumulative" tier. IE: If a user account has sponsored above x amount counting both monthly and one time sponsors.

I'm available for hire.
